### PR TITLE
Rename coordinate transform builder to transform map builder

### PR DESCRIPTION
**Pull request by krzysz00**
This makes the code less confusing

Fixes https://github.com/ROCmSoftwarePlatform/llvm-project-private/issues/483

### CODE REVIEW

File: mlir/include/mlir/Dialect/MIOpen/MIOpen.h [hunk 1]
```diff
@@ -83,14 +83,14 @@ namespace miopen {
 /// shape down or the lower shape up, such as passThrough() or pad(), "start" is
 /// the coordinate space you began building from and end is the coordinate space
 /// you're building towards (whether that's the upper or lower coordinates)
-class CoordTransformsBuilder {
+class TransformMapBuilder {
```
Comment (sjw36) [inline mlir/include/mlir/Dialect/MIOpen/MIOpen.h]: Move to it's own file. Many things that include MIOpen do not need this.

**Review by sjw36 — COMMENTED** [review]
Nice. Fewer characters to type.. except mostly using the derived classes anyway.. :)

Please move this out of MIOpen.h.

File: mlir/include/mlir/Dialect/MIOpen/MIOpen.h (hunk 1 above)
Comment (krzysz00) [reply mlir/include/mlir/Dialect/MIOpen/MIOpen.h]: Yeah, that's a reasonable ask since we're already here

Comment (krzysz00) [reply mlir/include/mlir/Dialect/MIOpen/MIOpen.h]: Done

**Review by sjw36 — APPROVED** [review]
Looks good.

**Pull request merged**